progress/upload_status.py

```diff
@@ -69,40 +54,38 @@ def get_directory_status(directory, sample_sheet):
         data = reader.read().decode()
     info_file = json.loads(data)
     status = info_file[STATUS_FIELD]
-    if status in DIRECTORY_STATUS_LIST:
+    if status in DirectoryStatus.VALID_STATUS_LIST:
         result.status = status
     else:  # the status found in the file is not in the defined list
         raise exceptions.DirectoryError("Invalid Status in status file", directory)
 
     return result
 
 
-def write_directory_status(directory, status, run_id=None):
+def write_directory_status(directory_status, run_id=None):
```

Comment:
When writing status to file, we now use the `DirectoryStatus` object. This mirrors returning a `DirectoryStatus` object when reading from the status file.